compiler/qsc_formatter/src/main.rs

```diff
@@ -1,17 +1,195 @@
+                return; // don't count file in file_count
+            }
+        };
+        if self.is_write {
+            let formatted = format_str(&file_as_string);
+            if file_as_string != formatted {
+                match std::fs::write(path, formatted) {
+                    Ok(_) => {
+                        self.changed_files.push(path.display().to_string());
+                    }
+                    Err(e) => {
+                        println!("\t{Skip}Could not write to file: {e}{Reset}");
+                        self.skipped_files.push(path.display().to_string());
+                        return; // don't count file in file_count
+                    }
+                }
+            }
+        } else if !calculate_format_edits(&file_as_string).is_empty() {
+            self.changed_files.push(path.display().to_string());
+        }
+        self.file_count += 1;
+    }
+}
+
+enum OutputFormatting {
+    Error,
+    Skip,
+    Passing,
+    Reset,
+}
+
+impl Display for OutputFormatting {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let output = match self {
+            OutputFormatting::Error => "\x1B[1;91m", // Bold, Bright Red
+            OutputFormatting::Passing => "\x1B[1;92m", // Bold, Bright Green
+            OutputFormatting::Skip => "\x1B[1;93m",  // Bold, Bright Yellow
+            OutputFormatting::Reset => "\x1B[0m",
+        };
+        write!(f, "{output}")
+    }
+}
+
+fn main() -> Result<(), String> {
+    use OutputFormatting::*;
+
+    let mut file_walker = FileWalker::new();
+    if !file_walker.root.exists() {
+        return Err("Given path can not found.".to_string());
+    }
+    if !file_walker.root.is_dir() && !is_path_qs(&file_walker.root) {
+        return Err("Given path is not a folder or Q# file.".to_string());
+    }
+
+    file_walker.format_from_root();
+
+    println!("Ran against {} files.", file_walker.file_count);
+
+    if !file_walker.skipped_files.is_empty() {
+        println!("{Skip}Skipped {} files:", file_walker.skipped_files.len());
+        for f in file_walker.skipped_files.iter() {
+            println!("\t{f}");
+        }
+        print!("{Reset}");
+    }
+
+    if file_walker.is_write {
+        println!("Updated {} files:", file_walker.changed_files.len());
+        for f in file_walker.changed_files.iter() {
+            println!("\t{f}");
+        }
+        Ok(())
+    } else if !file_walker.changed_files.is_empty() {
+        println!(
+            "{Error}{} files are in need of formatting:",
+            file_walker.changed_files.len()
+        );
+        for f in file_walker.changed_files.iter() {
+            println!("\t{f}");
+        }
+        println!("Run the formatter to correct formatting for the above files.{Reset}");
+        Err("Files are not formatted.".to_string())
+    } else if !file_walker.skipped_files.is_empty() {
+        println!(
```

Comment:
Is there a good reason we'd fail to read files? Shouldn't the command fail if we had to skip files?

Reply:
It didn't seem very ergonomic to me to stop the whole formatting if one of the files encountered couldn't be read from due to permissions.

Reply:
Or do you mean the formatter should still run all the way through but return an error code?

Reply:
done